src/aap_eda/wsapi/consumers.py

```diff
@@ -417,3 +435,37 @@ def get_eda_system_vault_passwords(
             )
 
         return vault_passwords
+
+    def _get_url(self, message: ActionMessage, inputs: dict) -> str:
+        if message.action not in ("run_job_template", "run_workflow_template"):
+            return ""
+        url = message.url
+        if not message.controller_job_id:
+            return url
+
+        api_url = inputs["host"]
+        urlparts = urlparse(api_url)
+
+        path = urlparts.path.rstrip("/")
+        if path == "":
+            path = "/"
+        if path in settings.API_PATH_TO_UI_PATH_MAP:
+            path = settings.API_PATH_TO_UI_PATH_MAP[path]
+
+        if message.action == "run_job_template":
+            slug = f"{path}/jobs/playbook/{message.controller_job_id}/details/"
+        elif message.action == "run_workflow_template":
+            slug = f"{path}/jobs/workflow/{message.controller_job_id}/details/"
```

Comment:
Should be `if/else`.  The check at entry guarantees it's one or the other.
As written it gives the impression that `slug` may not be set before use in `urlunparse`.